### PR TITLE
Fix git deps with branches vendoring

### PR DESCRIPTION
**Pull request by Ericson2314**
The branch name extraction logic was broken.

As a result of this, we can stop pregenerating the test for this case, demonstrating that `tools.nix` handles it.

### CODE REVIEW

**Comment by Ericson2314** [comment]
@balsoft would you mind reviewing this, since I think you contributed large parts of the code I was modifying?

File: tools.nix [hunk 1]
```diff
@@ -232,11 +249,12 @@ rec {
...
           let
             parsed = parseGitSource source;
-            src = builtins.fetchGit {
+            src = builtins.fetchGit ({
               submodules = true;
               inherit (parsed) url rev;
-              ref = attrs.branch or "master";
-            };
+            } // lib.optionalAttrs (parsed ? branch) {
+              ref = parsed.branch;
```
Comment (balsoft) [inline tools.nix:256]: Is the branch always set in the git source URL?

Comment (Ericson2314) [reply tools.nix:256]: It isn't, in which case it just takes the default one (remote side `HEAD`). I say this matter for some `main` vs `master` defaults.

Comment (balsoft) [reply tools.nix:256]: Yeah, upon a second reading it makes sense.

**Review by balsoft — APPROVED** [review]

**Comment by Ericson2314** [comment]
Thanks!

**Pull request merged**